include/nbl/config/BuildConfigOptions.h.in

```diff
@@ -106,6 +106,22 @@
     #define NBL_API2
 #endif
 
+// disabling 3rdparty warnings
+#if defined(NBL_SILENT_WARNINGS_WITH_WRAPPER)
+    #if defined(_MVC_VER)
+        #define NBL_PUSH_DISABLE_WARNINGS __pragma(warning(push, 0))
+        #define NBL_POP_DISABLE_WARNINGS __pragma(warning(pop))
+    #endif
+
+    #if !defined(NBL_PUSH_DISABLE_WARNINGS)
```

Comment:
now you have a bug, those need to go outside `#if defined(NBL_SILENT_WARNINGS_WITH_WRAPPER)` check

```c++
#if defined(NBL_SILENT_WARNINGS_WITH_WRAPPER)
  #if defined(_MVC_VER)
    #define NBL_PUSH_DISABLE_WARNINGS __pragma(warning(push, 0))
    #define NBL_POP_DISABLE_WARNINGS __pragma(warning(pop))
  // TODO: #elseif macro definitions for unix compilers
  #endif
#endif

#if !defined(NBL_PUSH_DISABLE_WARNINGS)
  #define NBL_PUSH_DISABLE_WARNINGS
#endif

#if !defined(NBL_POP_DISABLE_WARNINGS)
  #define NBL_POP_DISABLE_WARNINGS
#endif
```